backend/src/channels/channels.service.ts

```diff
@@ -0,0 +1,182 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Socket } from 'socket.io';
+import { User } from 'src/users/entities/user.entity';
+import { UsersService } from 'src/users/users.service';
+import { Repository } from 'typeorm';
+import { Channel } from './entities/channel.entity';
+import { ChannelType } from './entities/channel.entity';
+
+@Injectable()
+export class ChannelsService {
+  constructor(
+    @InjectRepository(Channel)
+    private readonly channelRepository: Repository<Channel>,
+    @InjectRepository(User) private readonly userRepository: Repository<User>,
+    private readonly userService: UsersService,
+  ) {}
+
+  async findAll(): Promise<Channel[]> {
+    console.log('This action returns all channels');
+    const channels = await this.channelRepository
+      .createQueryBuilder('channel')
+      .leftJoinAndSelect('channel.owner', 'owner')
+      .leftJoinAndSelect('channel.users', 'users')
+      .leftJoinAndSelect('channel.admins', 'admins')
+      .leftJoinAndSelect('channel.mutedUsers', 'mutedUsers')
+      .leftJoinAndSelect('channel.invitedUsers', 'invitedUsers')
+      .leftJoinAndSelect('channel.bannedUsers', 'bannedUsers')
+      .getMany();
+    return channels;
+  }
+
+  async getChannelById(id: string): Promise<Channel | undefined> {
+    try {
+      const channel = await this.channelRepository
+        .createQueryBuilder('channel')
+        .where('channel.id = :id', { id: id })
+        .leftJoinAndSelect('channel.owner', 'owner')
+        .leftJoinAndSelect('channel.users', 'users')
+        .leftJoinAndSelect('channel.admins', 'admins')
+        .leftJoinAndSelect('channel.mutedUsers', 'mutedUsers')
+        .leftJoinAndSelect('channel.invitedUsers', 'invitedUsers')
+        .leftJoinAndSelect('channel.bannedUsers', 'bannedUsers')
+        .getOne();
+      if (channel === null) {
+        throw new NotFoundException('Channel not found');
+      } else {
+        return channel;
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  async getChannelByName(name: string): Promise<Channel | undefined> {
+    try {
+      const channel = await this.channelRepository
+        .createQueryBuilder('channel')
+        .where('channel.name = :name', { name: name })
+        .leftJoinAndSelect('channel.owner', 'owner')
+        .leftJoinAndSelect('channel.users', 'users')
+        .leftJoinAndSelect('channel.admins', 'admins')
+        .leftJoinAndSelect('channel.mutedUsers', 'mutedUsers')
+        .leftJoinAndSelect('channel.invitedUsers', 'invitedUsers')
+        .leftJoinAndSelect('channel.bannedUsers', 'bannedUsers')
+        .getOne();
+      if (channel === null) {
+        throw new NotFoundException('Channel not found');
+      } else {
+        return channel;
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  async findVisibleChannelsWithoutUser(
```

Comment:
Ich hätte eigentlich lieber ne andere Query, und zwar alle Channel die für einen User visible sind, inkl. denen wo er drin ist - weil das sind alle die ich darstellen will, und dann kann ich ja über die Channel-Liste im User prüfen ob er drin ist oder nicht...

Außerdem fehlt mir hier noch private Channel, in die der User eingeladen ist - die würde ich auch mit anzeigen wollen.